core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -106,6 +106,9 @@
         Setting.byteSizeSetting("index.translog.flush_threshold_size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    public static final Setting<TimeValue> INDEX_SEQ_NO_CHECKPOINT_SYNC_INTERVAL =
+        Setting.timeSetting("index.seq_no.checkpoint_sync_interval", new TimeValue(30, TimeUnit.SECONDS),
+            new TimeValue(-1, TimeUnit.MILLISECONDS), Property.Dynamic, Property.IndexScope);
```

Comment:
Okay, I see what you mean. As interval tends to zero, the frequency of updates increases until zero when there are none. You're right it would be better in this case to disallow zero.